crypto/evp/digest.c

```diff
@@ -182,6 +182,7 @@ static int evp_md_init_internal(EVP_MD_CTX *ctx, const EVP_MD *type,
 #endif
 
     EVP_MD_CTX_clear_flags(ctx, EVP_MD_CTX_FLAG_CLEANED);
+    EVP_MD_CTX_clear_flags(ctx, EVP_MD_CTX_FLAG_FINALISED);
```

Comment:
Just OR the flags should work, I think

Reply:
Seemed clearer this way, but will change if it is a deal breaker :)